storage/series/types.go

```diff
@@ -251,4 +253,29 @@ type Options interface {
 
 	// IdentifierPool returns the identifierPool
 	IdentifierPool() ident.Pool
+
+	// SetStats sets the configured Stats.
+	SetStats(value Stats) Options
+
+	// Stats returns the configured Stats.
+	Stats() Stats
+}
+
+// Stats is passed down from above to avoid constructing tally constructs per series/buffer.
+type Stats struct {
+	EncoderCreated tally.Counter
+}
+
+// NewStats returns a new Stats for the provided scope.
+func NewStats(scope tally.Scope) Stats {
+	return Stats{
+		EncoderCreated: scope.Counter("encoder-created"),
+	}
+}
+
+// IncCreatedEncoders incs the EncoderCreated stat.
+func (s Stats) IncCreatedEncoders() {
+	if int(time.Now().UnixNano())%100 == 0 {
```

Comment:
Don't we have the ability to set sampling in tally itself?

Reply:
Also if this method is present, do you want to still expose the raw EncoderCreated field? its public right now